pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -3405,6 +3494,350 @@ def SeismicActivityExceptionRule(model, k, t):
     return model
 
 
+def pipeline_hydraulics(model, opt):
+
+    if model.config.hydraulics == Hydraulics.post_process:
+        """
+        For the post process method, the pressure is computed using the
+        Hazen-Williams equation in a separate stand alone model. This model
+        consists of a pressure balance equation at each node and bounds for
+        pressure. The objective is to minimize the total pressure change (deltaP)
+        due to pumps and valves.
+        """
+        # Create a model and add hydraulics constraint to compute pressure at each node
+        mh = ConcreteModel()
+
+    elif model.config.hydraulics == Hydraulics.co_optimize:
+        """
+        For the co-optimize method, first, the strategic model is cloned and then,
+        constraints for computing pressures at every node using
+        Hazen-Williams equation is added to the model. The original objective
+        function of minimizing the cost is modified to include of the operating
+        cost of pumps and valves as a function of the deltaP.
+        """
+        # Clone the model to add hydraulics constraints
+        # The cloned model will be a MINLP formulation. The exisiting objective
+        # must be modified to add operating costs for pumps and valves.
+        mh = model.clone()
+        del mh.objective
+
+    # declaring variables and parameters required regardless of the hydraulics method
+    mh.p_HW_material_factor_pipeline = Param(
+        initialize=130,
+        mutable=True,
+        units=pyunits.dimensionless,
+        doc="Pipeline material factor for Hazen-Williams equation",
+    )
+    mh.v_PumpHead = Var(
+        model.s_LLA,
+        model.s_T,
+        within=NonNegativeReals,
+        initialize=0,
+        units=pyunits.meter,
+        doc="Pump Head added in the direction of flow, m",
+    )
+    mh.v_ValveHead = Var(
+        model.s_LLA,
+        model.s_T,
+        within=NonNegativeReals,
+        initialize=0,
+        units=pyunits.meter,
+        doc="Valve Head removed in the direction of flow, m",
+    )
+    mh.p_rhog = Param(
+        initialize=9.8 * 1000,
+        units=model.model_units["pressure"] / pyunits.meter,
+        doc="g (m/s2) * density of PW (assumed to be 1000 kg/m3)",
+    )
+    mh.p_PumpingCost = Param(
+        initialize=1.2,
+        doc="Cost of pumping water",
+    )
```

Comment:
Why 1.2? Should the user be allowed to modify/specify this parameter? As of now, it appears there isn't a way for the user to change it away from 1.2.

Reply:
This parameter is now replaced by more granular and mutable parameters like electricity price, and efficiencies

Reply:
Resolving.